frontend/src/actions/plans.ts

```diff
@@ -0,0 +1,143 @@
+"use server";
+
+import { revalidatePath } from "next/cache";
+
+import { auth, fetchToAdonis } from "@/lib/auth";
+
+interface CreatePlanResponseType {
+  success: boolean;
+  message: string;
+  schedule: {
+    name: string;
+    userId: number;
+    createdAt: string;
+    updatedAt: string;
+    id: number;
+  };
+}
+
+interface PlanResponseType {
+  name: string;
+  userId: number;
+  id: number;
+  createdAt: string;
+  updatedAt: string;
+  courses: Array<{
+    id: string;
+    name: string;
+    department: string;
+    lecturer: string;
+    type: string;
+    ects: number;
+    semester: number;
+    groups: Array<{
+      id: number;
+      name: string;
+      day: string;
+      time: string;
+      room: string;
+    }>;
+  }>;
+  registrations: Array<{
+    id: string;
+    name: string;
+  }>;
+}
+
+interface DeletePlanResponseType {
+  success: boolean;
+  message: string;
+}
+
+export const createNewPlan = async ({
+  name,
+  courses,
+  registrations,
+  groups,
+}: {
+  name: string;
+  courses: Array<{ id: string }>;
+  registrations: Array<{ id: string }>;
+  groups: Array<{ id: number }>;
+}) => {
+  const isLogged = await auth({});
```

Comment:
czemu tu sie nic nie przekazuje do autha?

i czemu funkcja auth się tak nazywa, a nie checkIsLoggedIn()?

Reply:
bo ona domyslnie zbiera cookiedy z headers() ale dodalem opcje zeby przekazac customowe jesli robie to w api route z requesta

Reply:
zmień nazwę i pozwól na brak argumentów, żeby nie przekazywać pustego obiektu

Reply:
ogolnie zamysl jest taki ze to bedzie zwracac ci usera jesli jestes zalogowany wiec po prostu z tymi argumentami zmienie, i bedzie albo null albo user, wdyt?